formula-android/src/main/java/com/instacart/formula/fragment/FragmentLifecycle.kt

```diff
@@ -1,65 +1,34 @@
 package com.instacart.formula.fragment
 
-import android.content.Context
 import androidx.fragment.app.Fragment
-import androidx.fragment.app.FragmentActivity
 import androidx.fragment.app.FragmentInspector
 import androidx.fragment.app.FragmentManager
-import com.instacart.formula.fragment.FragmentLifecycle.lifecycleEvents
 import com.instacart.formula.integration.LifecycleEvent
-import io.reactivex.Observable
-import io.reactivex.android.MainThreadDisposable
 
 /**
  * Provides utility method [lifecycleEvents] to track what fragments are added and removed from the backstack.
  */
-object FragmentLifecycle {
+internal object FragmentLifecycle {
 
-    fun shouldTrack(fragment: Fragment): Boolean {
+    internal fun shouldTrack(fragment: Fragment): Boolean {
         return !fragment.retainInstance && !FragmentInspector.isHeadless(fragment)
     }
 
-    private fun isKept(fragmentManager: FragmentManager, fragment: Fragment): Boolean {
+    internal fun isKept(fragmentManager: FragmentManager, fragment: Fragment): Boolean {
         return !fragment.isRemoving
     }
 
-    /**
-     * Must subscribe to the state before calling Activity.super.onCreate(),
-     * otherwise you might miss fragment event
-     */
-    fun lifecycleEvents(activity: FragmentActivity): Observable<FragmentLifecycleEvent> {
```

Comment:
Removing `Observable` based event logic and moving it into `FragmentFlowRenderView`.